### DISCUSSION TITLE
Couldn't find program: 'bash'

### DISCUSSION THREAD

**Discussion by Tangzl922** [pybamm-team/PyBaMM#2091] [Q&A] [answered]
Hi,
when I run the tutorial file for parameter management I got this error
![image](https://user-images.githubusercontent.com/73125328/172340765-61863c05-843c-429b-bf31-53646b130e3d.png)

I also read this issue https://github.com/pybamm-team/PyBaMM/issues/1697 and tried to use pybamm.parameters_cli.edit_parameter, but this still doesn't work.
![image](https://user-images.githubusercontent.com/73125328/172341334-0c8fd3e2-e0c3-4932-9bf1-ecbc7a18e042.png)
![image](https://user-images.githubusercontent.com/73125328/172342007-be77af65-a2ca-403f-b3c3-31c63c9d8352.png)

Here is my parameter path, maybe helpful for you. And my version is 22.5.
![image](https://user-images.githubusercontent.com/73125328/172341658-7e9833d5-29ce-46b5-8125-bf0bee29407e.png)

Regards

**Comment by brosaplanella** [comment]
The `parameters_cli` functions are not supported in Windows. You can still define the folder structure manually for the parameters to work.

I will transfer this to a discussion for the record.

**Reply by Tangzl922** [reply]
Thanks for the reply. It seems that 'bash' still doesn't work for me, so I have no idea how to edit the parameters without changing the default parameter set.

**Comment by Saransh-cpp** [comment, marked as answer]
You can install `git` with `git bash` (usually installed by default with `git`). This would allow your notebooks to access `bash` and run the commands using it.

Alternatively, you can use `%%cmd` instead of `%%bash`. Note that this would require changing the `bash` commands in the notebook to `cmd` commands. For instance -
```
%%bash
if [[ ! -d "lithium_ion" ]]; then
echo "Directory lithium_ion does not exist."
fi
```
would change to -
```
%%cmd
@echo off
if exist "lithium_ion" (
  echo Directory lithium_ion exists 
) else (
  echo Directory lithium_ion does not exist
)
```
or something similar.

I have pushed a notebook identical to the original one, but swapping the `bash` commands with `cmd` commands on my fork here - https://github.com/Saransh-cpp/PyBaMM/blob/parameter-management-in-windows/examples/notebooks/parameterization/parameter-management.ipynb. You can download this and run it locally on Windows without any errors!

I am not sure if these commands should be in some way displayed to the users in the original notebook. This issue should go away once the cli functions are exposed!

**Reply by Tangzl922** [reply]
Thanks for your answer, that really works for me!

**Reply by Nithin3297** [reply]
@Saransh-cpp , thanks for your help. I tried the windows version notebook you provided in this note book . I get the error message "pybamm_edit_parameter" is not recognised as an internal or external command, operable program or batch file. I think there is a path issue with my spyder. Kindly help me resolve this issue.

**Reply by Saransh-cpp** [reply]
Hi @Nithin3297. There have been considerable changes in PyBaMM since my comment, in fact, the parameter CLI has been deprecated. I would suggest that you go through the current documentation and ignore this notebook.
